src/models/tests/gamma_poisson_model_test.py

```diff
@@ -44,131 +44,136 @@ def gamma_poisson_pmf(k, alpha, beta):
+        kplus_actual = [3745, 933, 230, 55, 12, 2, 0, 0, 0, 0]
+        rp = ReachPoint([5000], kplus_actual)
+        gpm = GammaPoissonModel([rp])
+        Imax, N, alpha, beta = gpm._fit_histogram_chi2_distance(rp, nstarting_points=1)
+        h_predicted = gpm._expected_histogram(5000, Imax, N, alpha, beta)
         for i in range(len(h_actual)):
             self.assertTrue(
-                (h_actual[i] - h_predicted[i]) ** 2 < 1,
+                (h_actual[i] - h_predicted[i]) ** 2 / h_predicted[i] < 1,
                 f"Discrepancy found at position {i}. "
                 f"Got {h_predicted[i]} Expected {h_actual[i]}",
             )
 
     def test_fit_histogram_fixed_N(self):
-        gpm = GammaPoissonModel([ReachPoint([], [])])
-        I0, N0, alpha0, beta0 = 4000, 10000, 1, 1
-        h_actual = gpm._expected_histogram(I0, I0 / 0.1, N0, alpha0, beta0)
-        Imax, alpha, beta = gpm._fit_histogram_fixed_N(h_actual, N0)
-        Ih = np.sum([h_actual[i] * (i + 1) for i in range(len(h_actual))])
-        h_predicted = gpm._expected_histogram(Ih, Imax, N0, alpha, beta)
+        # The following point corresponds to
+        #  I, Imax, N, alpha0, beta0 = 15000, 30000, 10000, 1.0, 2.0
+        # This represents a distribution with mean = 3 and variance = 6.
+        h_actual = [3749, 1875, 937, 468, 234, 117, 58, 29, 14, 7]
+        kplus_actual = [7488, 3739, 1864, 927, 459, 225, 108, 50, 21, 7]
+        rp = ReachPoint([15000], kplus_actual)
+        gpm = GammaPoissonModel([rp])
+        Imax, alpha, beta = gpm._fit_histogram_fixed_N(rp, 10000)
+        h_predicted = gpm._expected_histogram(15000, Imax, 10000, alpha, beta)
         for i in range(len(h_actual)):
             self.assertTrue(
-                (h_actual[i] - h_predicted[i]) ** 2 < 1,
+                (h_actual[i] - h_predicted[i]) ** 2 / h_predicted[i] < 1,
                 f"Discrepancy found at position {i}. "
                 f"Got {h_predicted[i]} Expected {h_actual[i]}",
             )
 
-    def test_fit_variable_N(self):
+    @patch(
+        "wfa_planning_evaluation_framework.models.gamma_poisson_model.GammaPoissonModel._fit_histogram_chi2_distance"
+    )
+    def test_fit_variable_N(self, mock_gamma_poisson_model):
+        mock_gamma_poisson_model.return_value = (30000, 10000, 1.0, 2.0)
         h_actual = [2853, 813, 230, 64, 17, 4, 1, 0, 0, 0]
         rp = ReachPoint([4000], h_actual)
         gpm = GammaPoissonModel([rp])
         gpm._fit()
-        self.assertAlmostEqual(gpm._max_reach, 9500, delta=100)
-        self.assertAlmostEqual(gpm._alpha, 1.12, delta=0.05)
-
-    def test_fit_fixed_N(self):
+        self.assertAlmostEqual(gpm._max_reach, 10000, delta=1)
+        self.assertAlmostEqual(gpm._alpha, 1.0, delta=0.01)
+
+    @patch(
+        "wfa_planning_evaluation_framework.models.gamma_poisson_model.GammaPoissonModel._fit_histogram_fixed_N"
+    )
+    def test_fit_histogram_fixed_N(self, mock_gamma_poisson_model):
```

Comment:
There is another method above with exactly the same name `test_fit_histogram_fixed_N`

Reply:
Thanks for catching this!